src/extension/coqPilot.ts

```diff
@@ -81,70 +75,126 @@ export class CoqPilot {
+            this.sessionExtensionState.dispose();
+            this.healthStatusIndicator.updateStatusBar(false);
+        } else {
+            this.sessionExtensionState = await SessionExtensionState.create(
+                this.globalExtensionState.logOutputChannel,
+                this.globalExtensionState.eventLogger
+            );
+            this.globalExtensionState.hasActiveSession = true;
+            this.healthStatusIndicator.updateStatusBar(true);
+        }
     }
 
     private async performSpecificCompletionsWithProgress(
         shouldCompleteHole: (hole: ProofStep) => boolean,
-        editor: TextEditor
+        editor: TextEditor,
+        abortSignal: AbortSignal
     ) {
-        await window.withProgress(
-            {
-                location: ProgressLocation.Window,
-                title: `${pluginName}: In progress`,
-            },
-            async () => {
-                try {
-                    await this.performSpecificCompletions(
-                        shouldCompleteHole,
-                        editor
-                    );
-                } catch (e) {
-                    if (e instanceof SettingsValidationError) {
-                        e.showAsMessageToUser();
-                    } else if (e instanceof CoqLspStartupError) {
-                        showMessageToUserWithSettingsHint(
-                            EditorMessages.coqLspStartupFailure(e.path),
-                            "error",
-                            `${pluginId}.coqLspServerPath`
-                        );
-                    } else {
-                        showMessageToUser(
-                            e instanceof Error
-                                ? EditorMessages.errorOccurred(e.message)
-                                : EditorMessages.objectWasThrownAsError(e),
-                            "error"
-                        );
-                        console.error(buildErrorCompleteLog(e));
-                    }
-                }
+        if (!this.globalExtensionState.hasActiveSession) {
+            showMessageToUser(EditorMessages.extensionIsPaused, "warning");
+            return;
+        }
+        try {
+            this.healthStatusIndicator.showSpinner();
+
+            await this.performSpecificCompletions(
+                shouldCompleteHole,
+                editor,
+                abortSignal
+            );
+        } catch (e) {
+            if (e instanceof SettingsValidationError) {
+                e.showAsMessageToUser();
+            } else if (e instanceof CoqLspStartupError) {
+                showMessageToUserWithSettingsHint(
+                    EditorMessages.coqLspStartupFailure(e.path),
+                    "error",
+                    `${pluginId}.coqLspServerPath`
+                );
+            } else if (
+                e instanceof CompletionAbortError ||
+                !this.globalExtensionState.hasActiveSession
```

Comment:
Could you, please, explain what this check means here? Maybe in the code in the form of the comment also 👉👈